drivers/syslog/syslog.h

```diff
@@ -72,17 +100,19 @@ EXTERN FAR const struct syslog_channel_s *g_syslog_channel
  *   SYSLOG device.  That would be a good extension.
  *
  * Input Parameters:
- *   devpath - The full path to the character device to be used.
- *   oflags  - File open flags
- *   mode    - File open mode (only if oflags include O_CREAT)
+ *   channel    - Handle to syslog channel to be used.
+ *   devpath    - The full path to the character device to be used.
+ *   oflags     - File open flags.
+ *   mode       - File open mode (only if oflags include O_CREAT).
  *
  * Returned Value:
  *   Zero (OK) is returned on success; a negated errno value is returned on
  *   any failure.
  *
  ****************************************************************************/
 
-int syslog_dev_initialize(FAR const char *devpath, int oflags, int mode);
+int syslog_dev_initialize(FAR struct syslog_channel_s *channel,
```

Comment:
let syslog_dev_initialize allocate syslog_channel_s(syslog_dev_s) internally instead

Reply:
Yes I followed this idea. It is indeed better now.